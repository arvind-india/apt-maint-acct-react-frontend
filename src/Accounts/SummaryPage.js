import React from 'react'
import { connect } from 'react-redux'
import { Table } from 'reactstrap'

import { FlashMessage } from '../_components'
import { accountActions as actions, userActions } from '../_actions'


export class Summary extends React.Component {

  componentDidMount() {
    // this.props.dispatch(actions.getSummaryList())
    // this.props.dispatch(userActions.getAll())
    this.props.getSummaryList()
    this.props.getAll()
  }

  render() {
    const { summaries, alert, users } = this.props
    let models = summaries
    return (
      <div>
        <h3>Accounts Summary</h3>
        {alert.message && <FlashMessage text={alert.message} delay={5000}/>}
        {models.items && users.items && this.showList() }
      </div>
    )
  }

  showList(){
    const { summaries } = this.props
    let models = summaries
    return <Table>
      <thead>{ this.headerRow() }</thead>
      <tbody>
        { models.items.map((model, index) => this.bodyRow(model,index)) }
      </tbody>
    </Table>
  }

  headerRow() {
    return <tr>
      <th>#</th>
      <th>Year-Month</th>
      <th>Collections</th>
      <th>Expenses</th>
      <th>Difference</th>
      <th>Cumulative Difference</th>
    </tr>
  }

  bodyRow(model,index) {
    return <tr key={model.id}>
      <td>{index+1}</td>
      <td>{model.yr_mo}</td>
      <td>{model.cr}</td>
      <td>{model.dr}</td>
      <td>{model.diff}</td>
      <td>{model.cumulativeDiff}</td>
    </tr>
  }

}

function mapStateToProps(state) {
  const { summaries, alert, users } = state
  return {
    summaries,
    alert,
    users
  }
}

function mapDispatchToProps(dispatch) {
  return {
    getSummaryList: () => {
      dispatch(actions.getSummaryList())
    },
    getAll: () => {
      dispatch(userActions.getAll())
    }
  }
}

//const connectedSummaryPage = connect(mapStateToProps)(SummaryPage)
//export { connectedSummaryPage as SummaryPage }
export default connect(mapStateToProps, mapDispatchToProps)(Summary)
