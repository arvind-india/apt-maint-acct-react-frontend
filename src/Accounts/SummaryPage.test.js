// SummaryPage.test.js

import React from 'react'
import { shallow } from 'enzyme'

import { Summary } from './SummaryPage'
const mockSummaries = [
  {
    id: 1,
    yr_mo: '2018-04',
    cr: 100,
    dr: 0,
    diff: 100,
    cumulativeDiff: 100
  },
  {
    id: 2,
    yr_mo: '2018-04',
    cr: 100,
    dr: 0,
    diff: 100,
    cumulativeDiff: 200
  },
  {
    id: 3,
    yr_mo: '2018-04',
    cr: 0,
    dr: 50,
    diff: -50,
    cumulativeDiff: 150
  },
]
const mockProps = {
  summaries: {
    items: mockSummaries
  },
  users: {
    items: []
  },
  alert: {message: "testingAlert"},
  authzn: {
    allowsAdd: true,
    allowsEdit: true,
    allowsDelete: true,
    condition: ''
  },
  getAll: jest.fn(),
  getSummaryList: jest.fn()
}

const component = shallow(<Summary {...mockProps} />)

describe('SummaryPage params test', () => {
  it('should render SummaryPage without crashing', () => {
    component
  })

  it('should render initial layout', () => {
    expect(component.getElements()).toMatchSnapshot()
  })
})

describe('SummaryPage rows test', () => {
  let numberOfMockSummary = mockProps.summaries.items.length
  let tbody = component.find('tbody')
  let rows = tbody.find('tr')
  it('should have '+numberOfMockSummary+' rows', () => {
    expect(rows.length).toEqual(numberOfMockSummary)
  })
})
