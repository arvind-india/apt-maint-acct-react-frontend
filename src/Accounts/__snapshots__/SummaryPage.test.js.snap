// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryPage params test should render initial layout 1`] = `
Array [
  <div>
    <h3>
        Accounts Summary
    </h3>
    <FlashMessage
        delay={5000}
        text="testingAlert"
    />
    <Table
        responsiveTag="div"
        tag="table"
    >
        <thead>
            <tr>
                <th>
                    #
                </th>
                <th>
                    Year-Month
                </th>
                <th>
                    Collections
                </th>
                <th>
                    Expenses
                </th>
                <th>
                    Difference
                </th>
                <th>
                    Cumulative Difference
                </th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td>
                    1
                </td>
                <td>
                    2018-04
                </td>
                <td>
                    100
                </td>
                <td />
                <td>
                    100
                </td>
                <td>
                    100
                </td>
            </tr>
            <tr>
                <td>
                    2
                </td>
                <td>
                    2018-04
                </td>
                <td>
                    100
                </td>
                <td />
                <td>
                    100
                </td>
                <td>
                    200
                </td>
            </tr>
            <tr>
                <td>
                    3
                </td>
                <td>
                    2018-04
                </td>
                <td />
                <td>
                    50
                </td>
                <td>
                    -50
                </td>
                <td>
                    150
                </td>
            </tr>
        </tbody>
    </Table>
</div>,
]
`;
